src/CMakeLists.txt

```diff
@@ -224,6 +226,8 @@ set(utils_files
     nupic/utils/VectorHelpers.hpp
     nupic/utils/SdrMetrics.cpp
     nupic/utils/SdrMetrics.hpp
+    nupic/utils/MurmurHash3.hpp
+    nupic/utils/MurmurHash3.cpp
```

Comment:
I don't want to download the official github repo for murmurhash3.  See comment https://github.com/htm-community/nupic.cpp/pull/278#issuecomment-471025667

I had to modify these files to work.  I replaced a switch-case statement with several if statements, because of the compiler flag `-Werror=implicit-fallthrough`  I also replaced all of the types such as `uint32_t` with our types `UInt32`

Reply:
ok, that sounds reasonable, I had forgotten that comment. 
Still, please move to external/

> I also removed the all hash functions except the one we are going to use (32-bit hash for x86 architecture)

we might eventually use ARM too, #131 , but that is not a blocker for now, and hopefully will just compile this as well.

Reply:
It should compile just fine.  This hash is optimized to run on 32 bit processors (x86) as opposed to 64 bit processors (x86_64).  It should be versatile across all of the platforms.

Reply:
> please move [murmurhash3] external/

I agree that it should go there, since it does not use the same license as Nupic.

Currently CMake includes but does not compile the code in `external/common/include` so I'm going to leave it where it is for now.